### PR TITLE
Support for multiple Max7219 units in a chain

### PR DESCRIPTION
**Pull request by Roxy-3D**
I'll move this support over to bugfix_2.0.0 in the next few days.   

And then look at updating the Max7219_idle() routine to make use of multiple
units in a chain.

### CODE REVIEW

File: Marlin/Max7219_Debug_LEDs.h [hunk 1]
```diff
@@ -69,18 +76,72 @@ void Max7219_LED_On(const uint8_t x, const uint8_t y);
...
+#ifndef MAX7219_ROTATE
+  #define MAX7219_ROTATE 0
+#endif
+#define _ROT ((MAX7219_ROTATE + 360) % 360)
+#if _ROT == 0
+  #define MAX7219_UPDATE_AXIS     y   // Fast line update axis for this orientation of the matrix display
+  #define MAX7219_X_LEDS          (8 * MAX7219_NUMBER_UNITS)
+  #define MAX7219_Y_LEDS          8
+  #define XOR_7219(x, y)          LEDs[y + (x >> 3) * 8] ^= _BV(7 - (x & 0x07))
+  #define SET_PIXEL_7219(x, y)    LEDs[y + (x >> 3) * 8] |= _BV(7 - (x & 0x07))
```
Comment (thinkyhead) [inline Marlin/Max7219_Debug_LEDs.h:110]: To get slightly fewer shifts, `(x >> 3) * 8` is the same as `((x >> 3) << 3)` is the same as `(x & 0xF8)`.

**Comment by thinkyhead** [comment]
You figured it out. Awesome.

File: Marlin/Max7219_Debug_LEDs.h (hunk 1 above)
Comment (Roxy-3D) [reply Marlin/Max7219_Debug_LEDs.h:110]: Yeah...   I'm starting to trust the compiler more and more.   Usually on this stuff...  GCC is smart enough and does something pretty reasonable.

**Comment by Roxy-3D** [comment]
I'm in the middle of printing a 2 unit bracket for my printer.   So that is keeping me from doing the Bugfix_2.0.0 version.   But that will happen soon enough.    I'm going to have to wire stuff up, so that will delay things for a day or two.     But with the display mounted and wired up...  I'll be able to test things and start working on the 16x8 Max7219_idle() functions.  (I'll get bugfix_2.0.0 to match bugfix_1.1.x functionality first.)

So like all software projects...   The end date is slipping out a week.

**Pull request merged**